TypeSystem/Serialization/test.cc

```diff
@@ -211,6 +228,25 @@ TEST(Serialization, JSON) {
       "{\"j\":43,\"q\":\"bar\",\"v\":[\"one\",\"two\"],\"z\":{\"i\":42,\"s\":\"foo\",\"b\":true,\"e\":100}}",
       JSON(complex_object));
 
+  // Derived struct serialization.
+  DerivedSerializable derived_object;
+  derived_object.i = 48;
+  derived_object.s = "baz";
+  derived_object.b = true;
+  derived_object.e = Enum::SET;
+  derived_object.d = 3.14;
```

Comment:
The `.14` part is a bit scary for the unit test, since its denumerator is not a power of two, and one might get `0.139999999...` on a different version of RapidJSON. Why not `.25` or `.125` to stay on the safe side? :-)